packages/api-client/src/notification/NotificationAPI.ts

```diff
@@ -81,16 +81,30 @@ export class NotificationAPI {
    * @param lastNotificationId Only return notifications more recent than this
    * @see https://staging-nginz-https.zinfra.io/swagger-ui/#!/push/fetchNotifications
    */
-  public async getAllNotifications(clientId?: string, lastNotificationId?: string): Promise<Notification[]> {
+  public async getAllNotifications(
+    clientId?: string,
+    lastNotificationId?: string,
+  ): Promise<{notifications: Notification[]; missedNotification?: string}> {
     let notificationList: Notification[] = [];
 
-    const getNotificationChunks = async (clientId?: string, lastNotificationId?: string): Promise<Notification[]> => {
-      const {notifications, has_more} = await this.getNotifications(
-        clientId,
-        NOTIFICATION_SIZE_MAXIMUM,
-        lastNotificationId,
-      );
+    const getNotificationChunks = async (
+      clientId?: string,
+      lastNotificationId?: string,
+    ): Promise<{notifications: Notification[]; missedNotification?: string}> => {
+      let payload: NotificationList = {notifications: [], time: '0', has_more: false};
+      let hasMissedNotifications = false;
+      try {
+        payload = await this.getNotifications(clientId, NOTIFICATION_SIZE_MAXIMUM, lastNotificationId);
+      } catch (error) {
+        if ((error as any).response.data) {
```

Comment:
In case of error, we check that the response has data. Which means that the backend detected that we are trying to get a notification that it doesn't have, but still sends back all the notifications it has